doc/source/structures/misc/string.rst

```diff
@@ -0,0 +1,311 @@
+.. _string:
+
+String
+====
+
+A :struct:`String` is an immutable sequence of characters in kOS.
+
+Creating strings
+-------------------
+
+Unlike other structures, strings are created with a special syntax::
+
+    // Create a new string
+    SET s TO "Hello, Strings!".
+    
+
+Strings are immutable. This means, once a string has been created, it 
+can not be directly modified. However, new strings can be created out
+of existing strings. For example::
+
+    // Create a new string with "Hello" replaced with "Goodbye"
+    SET s TO "Hello, Strings!".
+    SET t TO s:REPLACE("Hello", "Goodbye").
```

Comment:
I think I'll also add an example of what *doesn't* work in addition to this example of what *does*, to help those unfamiliar with languages that use immutable strings understand how it works better.  (i.e. maybe show that `set s[0] to "A"` is the sort of thing we're talking about not working when we say "immutable".)